app/src/main/java/org/mifos/mobilebanking/presenters/LoginPresenter.java

```diff
@@ -178,22 +178,25 @@ private boolean isCredentialsValid(final String username, final String password)
 
         final Resources resources = context.getResources();
         final String correctUsername = username.replaceFirst("\\s++$", "").trim();
+        boolean usernameValid = true;
```

Comment:
@m-sameer Check your logic, no additional variable is required and no checks for passwords are added.

Reply:
@miPlodder If I don't use an additional variable for checking the username validity and return false directly the password **EditText** will not be checked. :confused: 
These are the minimal changes I can think of, can you suggest any?

Reply:
@m-sameer Then, don't return at that stage. Try to improve your solution.

Reply:
@miPlodder If I use a `TextWatcher` the credentials will be checked on every tap on the keyboard but they should be checked only when the login button is pressed. Also adding a listener will be more complicated and will add more lines than simply using a boolean variable.

Reply:
@m-sameer @miPlodder I have another approach which might satisfy need of both of you.

You can add a onFocus of Password EditText and then check if username is null or not.

Reply:
@luckyman20 That is what I'm talking. For example, if a user is entering his username, we don't want the validation/textwatcher to work. When he changes his focus to another field, then the textwatcher should be active. Now, when user edits his username, how the textwatcher will be active and showing all the relevant errors. 

See Paypal mobile interface while signing up.

Reply:
@miPlodder I think we won't be needing a TextWatcher for this, we have to split the `isCredentialsValid` into two separate methods namely `isUsernameValid` and `isPasswordValid` and we will call `isUsernameValid` in onFocus of Password EditText and then call `isPasswordValid` in `onSubmit`.